File: 7 kyu/Pair Zeros/solution.rb
def pair_zeros(arr)
  arr.join.gsub(/0(.*?)0/, '0\1').chars.map(&:to_i)
end
_____________________________________
def pair_zeros(arr)
  skip = true
  arr.reject{|el| el == 0 && skip = !skip}
end
_____________________________________
def pair_zeros a, t=nil
  a.select{|x|x!=0||t=!t}
end
_____________________________________
def pair_zeros(arr)
  zero = false
  arr.select {|n| n.zero? ? zero = !zero : true }
end
_____________________________________
def pair_zeros(arr)
  arr.join.gsub(/(0)([1-9]*)(0)/,'\1\2').chars.map(&:to_i)
end
_____________________________________
def pair_zeros(arr)
  pair = false
  result = []
  
  arr.each { |n|
    if n.zero? && pair
      pair = false
    else
      pair = true if n.zero?
      result << n
    end
  }
  
  result
end

File: 7 kyu/Pair Zeros/solution.js
function pairZeros(arr) {
  var seenZero = 0;
  return arr.filter(function(num){
    return num != 0 || seenZero++ % 2 == 0;
  });
}
_____________________________________
var pairZeros = (a, f) => a.filter(n => n || (f = !f));
_____________________________________
function pairZeros(a, f) {
  return a.filter(function(n) { return (n != 0 || (f = !f)); });
}
_____________________________________
const pairZeros = (arr, flag) =>
  arr.filter(val => val || (flag = !flag));
_____________________________________
var pairZeros = arr => 
  arr
    .join('')
    .replace(/0([^0]*)0/g, '0$1')
    .split('')
    .map(Number)
